test_scripts/Defects/5_0/1385_SDL_doesnt_deactivate_app_when_receives_OnEventChange_PHONE_CALL_from_HMI.lua

```diff
@@ -0,0 +1,115 @@
+
+-- [[ Required Shared libraries ]]
+local runner = require("user_modules/script_runner")
+local common = require("user_modules/sequences/actions")
+local connect = require("user_modules/dummy_connecttest")
+
+--[[ Test Configuration ]]
+runner.testSettings.isSelfIncluded = false
+
+--[[ Test params ]]
+local cases = {
+  navi_non_media = {
+    params = {
+      isMediaApplication = false,
+      appHMIType = { "NAVIGATION" }
+    },
+    expectedHmiStatus = {
+      hmiLevel = "LIMITED",
+      audioStreamingState = "NOT_AUDIBLE",
+      systemContext = "MAIN"
+    }
+  },
+  vcommunication_non_media = {
+    params = {
+      isMediaApplication = false,
+      appHMIType = { "COMMUNICATION" }
+    },
+    expectedHmiStatus = {
+      hmiLevel = "BACKGROUND",
+      audioStreamingState = "NOT_AUDIBLE",
+      systemContext = "MAIN"
+    }
+  },
+  default_media = {
+    params = {
+      isMediaApplication = true,
+      appHMIType = { "DEFAULT" }
+    },
+    expectedHmiStatus = {
+      hmiLevel = "BACKGROUND",
+      audioStreamingState = "NOT_AUDIBLE",
+      systemContext = "MAIN"
+    }
+  }
+}
+
+--[[Local Functions]]
+local function prepareConfig(params)
+  for key, value in pairs(params) do
+    config.application1.registerAppInterfaceParams[key] = value
+  end
+end
+
+local function sendOnEventChanged(testCase)
+  common.getHMIConnection():SendNotification("BasicCommunication.OnEventChanged", {
+    eventName = "PHONE_CALL",
+    isActive = true })
+  common.getMobileSession():ExpectNotification("OnHMIStatus", cases[testCase][expectedHmiStatus])
+  :Do(function()
+    common.getHMIConnection():SendNotification("BasicCommunication.OnEventChanged", {
+      eventName = "PHONE_CALL",
+      isActive = false })
+    end)
+  common.getMobileSession():ExpectNotification("OnHMIStatus", "FULL")
+end
+
+local function cleanSessions()
+  for i = 1, common.getAppsCount() do
+    connect.mobileSession[i]:Stop()
+    connect.mobileSession[i] = nil
+  end
+end
+
+local function unregisterApp()
+  local cid = common.getMobileSession():SendRPC("UnregisterAppInterface", {})
+  common.getHMIConnection():ExpectNotification("BasicCommunication.OnAppUnregistered",
+  { appID = common.getHMIAppId(), unexpectedDisconnect = false })
+  common.getMobileSession():ExpectResponse(cid, {success = true, resultCode = "SUCCESS"})
+  :Do(function()
+    cleanSessions()
+    end)
+end
+
+--[[ Scenario ]]
+runner.Title("Preconditions")
+runner.Step("Clean environment", common.preconditions)
+runner.Step("Start SDL, HMI, connect Mobile, start Session", common.start)
+
+for key, value in pairs(cases) do
+  runner.Title("TestCase: " .. key)
+  runner.Step("Prepare application params", prepareConfig, { value.params })
+  runner.Step("Register App1", common.registerAppWOPTU, { 1 })
+  runner.Step("Activate App1", common.activateApp, { 1 })
+
+  runner.Step("On event changed", sendOnEventChanged, { key })
```

Comment:
`sendOnEventChanged, { key }` causes error `PANIC: unprotected error in call to Lua API (./modules/atf/util.lua:124: bad argument #1 to 'pairs' (table expected, got string))`

Reply:
Fixed in sendOnEventChanged function. Thanks for the feedback.